### DISCUSSION TITLE
Unexpected method 'appcast' called on Cask cocoaspell.

### DISCUSSION THREAD

**Discussion by geipel** [Homebrew/discussions#5373] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8378cc825d83acffd125fb0fec041793df378a57
Last commit: 3 days ago
Core tap JSON: 16 May 23:33 UTC
Core cask tap JSON: 16 May 23:33 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 13.6.6-x86_64
CLT: 14.3.1.0.1.1683849156
Xcode: 15.2
```

### Output of `brew doctor`

```shell
Error: Unexpected method 'appcast' called on Cask cocoaspell.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Error: Unexpected method 'appcast' called on Cask cocoaspell.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  meld

Warning: You have the following deprecated, cask taps tapped:
  caskroom/cask
Untap them with `brew untap`.

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
  python-lxml

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-versions
Untap them with `brew untap`.

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/LibSideSyncOSX9.dylib
  /usr/local/lib/libaspell.15.dylib
  /usr/local/lib/libpspell.15.dylib
  /usr/local/lib/ss_conn_lib.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected header files:
  /usr/local/include/aspell.h
  /usr/local/include/pspell/pspell.h

Warning: Unbrewed '.la' files were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae and may need to be deleted.

Unexpected '.la' files:
  /usr/local/lib/libaspell.la
  /usr/local/lib/libpspell.la
```

### Description of issue

Most of the brew commands display a new error following a recent "brew upgrade". (Which I do regularly.)

    Unexpected method 'appcast' called on Cask cocoaspell.

Looking at "brew doctor": I also see a number of odd warnings about new "unbrewed" files that appear to be related.
How should I go about cleaning this up?

### Relevant casks

cocoaspell

**Comment by miccal** [comment, marked as answer]
The `cocoaspell` was removed a while ago: https://github.com/Homebrew/homebrew-cask/pull/138464.

Also read https://github.com/Homebrew/homebrew-cask/issues/173201#issuecomment-2101662209.

**Reply by geipel** [reply]
Oh, I see.
Curiously: uninstall worked to clear the error -- but left a lot of dangling files for me to clean up.

I'm also puzzled that if the formula was removed in 2022, why `brew upgrade` is only now complaining about it.

Regardless, this is now resolved.

Thanks!

**Reply by cho-m** [reply]
> I'm also puzzled that if the formula was removed in 2022, why `brew upgrade` is only now complaining about it.

The `appcast` DSL was deprecated since [`brew 4.1.0`](https://github.com/Homebrew/brew/releases/tag/4.1.0) and finally removed with latest [`brew 4.3.0`](https://github.com/Homebrew/brew/releases/tag/4.3.0). This is the general workflow for DSL removal (documented at https://docs.brew.sh/Releases).

**Reply by geipel** [reply]
Given my experience with trash files to clean up, it seems prudent that the full uninstall behavior should be available for a bit longer -- **after** this funky error makes it obvious (?) that an uninstall is required.

Speaking of which, I would hope that this error can be made more descriptive -- to explicitly say that the referenced cask is no longer supported.

Thanks folks, for your quick responses!

**Comment by richpav** [comment]
I'm getting the same error with blackhole but the cask has not been deleted.

```
$ brew info blackhole
Error: Unexpected method 'appcast' called on Cask blackhole.
Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
==> blackhole: 0.2.6
https://existential.audio/blackhole/
Installed
/usr/local/Caskroom/blackhole/0.2.6 (67KB)
==> Name
BlackHole
==> Description
None
==> Artifacts
BlackHole.v0.2.6.pkg (Pkg)
```

**Reply by cho-m** [reply]
Original `blackhole` cask was removed even longer ago https://github.com/Homebrew/homebrew-cask/pull/94528

You should replace it with one of the maintained channel-specific casks, e.g. [`blackhole-2ch`](https://formulae.brew.sh/cask/blackhole-2ch), [`blackhole-16ch`](https://formulae.brew.sh/cask/blackhole-16ch), or [`blackhole-64ch`](https://formulae.brew.sh/cask/blackhole-64ch)

**Reply by tueksta** [reply]
the answer is technically correct, but why can't homebrew be improved to show an appropriate warning, instead of a weird error message with a link to a bug reporter? it's not that unexpected that casks get removed, it should be handled gracefully.

**Comment by yordis** [comment]
For me it was:

```
Error: Unexpected method 'appcast' called on Cask adoptopenjdk8.
```

```
 Unexpected method 'appcast' called on Cask minikube.
 ```

**Reply by SoLongAndThanksForAllTheFish** [reply]
Ditto but for adoptopenjdk11

**Reply by dwiyatci** [reply]
For `adoptopenjdk*` (deprecated), do:
```
brew uninstall --cask adoptopenjdk8
brew uninstall --cask adoptopenjdk11
brew untap AdoptOpenJDK/openjdk

brew install --cask temurin@8
brew install --cask temurin@11
```

Ref:
https://github.com/AdoptOpenJDK/homebrew-openjdk

**Reply by subhashb** [reply]
It was `adoptopenjdk14` in my case, but the uninstall worked, and I no longer get the error message. Should we be installing the latest version of `adoptopenjdk11`? (16 is the latest).

**Reply by SMillerDev** [reply]
AdoptOpenJDK is deprecated, use Temurin casks
